src/nnfusion/engine/pass/graph/op_inplace_pass.cpp

```diff
@@ -47,6 +47,31 @@ bool OpInplacePass::run_on_graph(std::shared_ptr<Graph>& graph)
             auto op = std::dynamic_pointer_cast<GenericOp>(node->get_op_ptr());
             AddInplace(op, 0, 0, false);
         }
+
+        else if (node->get_op_type() == "Pad")
+        {
```

Comment:
Why this case is needed?

Reply:
This is for all kernels of pad op, instead of only pad antares kernel. It means if padding all zeros, pad op is marked as inplace op. And because the input and output are the same, memory manager will assign the same memory address for them. Then pad kernels can be eliminated. If we do not add inplace annotation in op_inplace_pass, only antares kernel can be eliminated, other kernels cannot.